pkgs/by-name/im/immich/machine-learning.nix

```diff
@@ -0,0 +1,85 @@
+{
+  lib,
+  src,
+  immich,
+  python3,
+}:
+python3.pkgs.buildPythonApplication {
+  pname = "immich-machine-learning";
+  inherit (immich) version;
+  src = "${src}/machine-learning";
+  pyproject = true;
+
+  postPatch = ''
+    substituteInPlace pyproject.toml --replace-fail 'fastapi-slim' 'fastapi'
+
+    # Allow immich to use pydantic v2
+    substituteInPlace app/schemas.py --replace-fail 'pydantic' 'pydantic.v1'
+    substituteInPlace app/main.py --replace-fail 'pydantic' 'pydantic.v1'
+    substituteInPlace app/config.py \
+      --replace-fail 'pydantic' 'pydantic.v1'
+  '';
+
+  pythonRelaxDeps = [
+    "setuptools"
+    "pydantic"
+  ];
+  pythonRemoveDeps = [ "opencv-python-headless" ];
+
+  build-system = with python3.pkgs; [
+    poetry-core
+    cython
+  ];
+
+  dependencies =
+    with python3.pkgs;
+    [
+      insightface
+      opencv4
+      pillow
+      fastapi
+      uvicorn
+      aiocache
+      rich
+      ftfy
+      setuptools
+      python-multipart
+      orjson
+      gunicorn
+      huggingface-hub
+      tokenizers
+      pydantic
+    ]
+    ++ uvicorn.optional-dependencies.standard;
+
+  doCheck = false;
+
+  postInstall = ''
+    mkdir -p $out/share
+    cp log_conf.json $out/share
```

Comment:
Does `immich` support reading this file from under a project-specific prefix? E.g. `$out/share/immich/log_conf.json`? This can be helpful when merging packages using `symlinkJoin` or `environment.systemPackages`